src/test/enginesynctest.cpp

```diff
@@ -84,54 +94,84 @@ class EngineSyncTest : public MockedEngineBackendTest {
     }
 
   private:
-    bool isMaster(const QString& group, bool explicitMaster) {
+    bool isMaster(const QString& group, SyncMode masterType) {
         if (group == m_sInternalClockGroup) {
-            if (!ControlObject::getControl(ConfigKey(m_sInternalClockGroup,
-                                                   "sync_master"))
-                            ->toBool()) {
+            if (double master = ControlObject::getControl(ConfigKey(m_sInternalClockGroup,
+                                                                  "sync_master"))
+                                        ->get();
+                    master != 1.0) {
+                qWarning() << "internal clock sync_master should be 1.0, is" << master;
                 return false;
             }
             if (m_pEngineSync->getMaster()) {
+                qWarning() << "no current master";
                 return false;
             }
             if (m_sInternalClockGroup != getMasterGroup()) {
-                return false;
-            }
-            // Internal Clock doesn't have explicit mode
-            if (explicitMaster) {
-                qWarning() << "test error, internal clock can never be explicit master";
+                qWarning() << "internal clock is not master, it's" << getMasterGroup();
                 return false;
             }
             return true;
         }
         if (group == m_sGroup1) {
             if (m_pEngineSync->getMaster() != m_pChannel1) {
+                qWarning() << "master pointer is wrong";
+                if (m_pEngineSync->getMaster()) {
+                    qWarning() << "it's " << m_pEngineSync->getMaster()->getGroup();
+                } else {
+                    qWarning() << "it's null";
+                }
```

Comment:
I suggest to use a ternary operator so will show up in the same log line

Reply:
done